src/components/WalletInit/RestoreWallet/WalletCredentialsScreen.js

```diff
@@ -273,9 +274,8 @@ class WalletCredentialsScreen extends React.Component<Props, State> {
       this.setState({isProcessing: false})
       navigation.navigate(ROOT_ROUTES.SHELLEY_WALLET)
     } catch (e) {
-      handleApiError(e, intl, 'Could not upgrade wallet')
-    } finally {
       this.setState({isProcessing: false})
+      handleApiError(e, intl, 'Could not upgrade wallet')
```

Comment:
This is a small side fix. I just removed the `finally` statement because when there is no error, the component is unmounted and `setState` would try to update the state of an unmounted component.